src/services/completions.ts

```diff
@@ -1256,7 +1283,11 @@ namespace ts.Completions {
                     /* delta */ 0,
                     { ...formatContext, options: formatOptions });
             });
-            return textChanges.applyChanges(syntheticFile.text, changes);
+
+            const allChanges = escapes
+                ? stableSort(concatenate(changes, escapes), (a, b) => compareTextSpans(a.span, b.span))

```

Comment:
Overlapping edits can defeat the sorting, but since the formatter deals only in whitespace and semicolons, and the characters we need to escape are only part of tokens, I don’t think it can ever be a problem in practice.

Reply:
I see, thanks.